ospf6d/ospf6_gr_helper.c

```diff
@@ -0,0 +1,1430 @@
+			vty_out(vty, "   ");
+			hash_walk(ospf6->ospf6_helper_cfg.enable_rtr_list,
+				  ospf6_print_vty_helper_dis_rtr_walkcb, vty);
+			vty_out(vty, "\n\n");
+		}
+
+		if (ospf6->ospf6_helper_cfg.last_exit_reason
+		    != OSPF6_GR_HELPER_EXIT_NONE) {
+			vty_out(vty, " Last Helper exit Reason :%s\n",
+				ospf6_exit_reason_desc
+					[ospf6->ospf6_helper_cfg
+						 .last_exit_reason]);
+
+			if (ospf6->ospf6_helper_cfg.active_restarter_cnt)
+				vty_out(vty,
+					" Number of Active neighbours in graceful restart: %d\n",
+					ospf6->ospf6_helper_cfg
+						.active_restarter_cnt);
+			else
+				vty_out(vty, "\n");
+		}
+
+
+	} else {
+		json_object_string_add(
+			json, "helperSupport",
+			(ospf6->ospf6_helper_cfg.is_helper_supported)
+				? "Enabled"
+				: "Disabled");
+		json_object_string_add(json, "strictLsaCheck",
+				       (ospf6->ospf6_helper_cfg.strict_lsa_check)
+					       ? "Enabled"
+					       : "Disabled");
+		json_object_string_add(
+			json, "restartSupoort",
+			(ospf6->ospf6_helper_cfg.only_planned_restart)
+				? "Planned Restart only"
+				: "Planned and Unplanned Restarts");
+
+		json_object_int_add(
+			json, "supportedGracePeriod",
+			ospf6->ospf6_helper_cfg.supported_grace_time);
+
+		if (ospf6->ospf6_helper_cfg.last_exit_reason
+		    != OSPF6_GR_HELPER_EXIT_NONE)
+			json_object_string_add(
+				json, "LastExitReason",
+				ospf6_exit_reason_desc
+					[ospf6->ospf6_helper_cfg
+						 .last_exit_reason]);
+
+		if (OSPF6_HELPER_ENABLE_RTR_COUNT(ospf6)) {
+			struct json_object *json_rid_array =
+				json_object_new_array();
+
+			json_object_object_add(json, "enabledRouterIds",
+					       json_rid_array);
+
+			hash_walk(ospf6->ospf6_helper_cfg.enable_rtr_list,
+				  ospf6_print_json_helper_dis_rtr_walkcb,
+				  json_rid_array);
+		}
+	}
+
+	if (detail) {
+		int cnt = 1;
+		struct listnode *i, *j, *k;
+		struct ospf6_area *oa;
+		json_object *json_neighbors = NULL;
+
+		for (ALL_LIST_ELEMENTS_RO(ospf6->area_list, i, oa))
+			for (ALL_LIST_ELEMENTS_RO(oa->if_list, j, oi)) {
+				struct ospf6_neighbor *nbr;
+
+				if (ospf6_interface_neighbor_count(oi) == 0)
+					continue;
+
+				if (uj) {
+					json_object_object_get_ex(
+						json, "Neighbors",
+						&json_neighbors);
+					if (!json_neighbors) {
+						json_neighbors =
+						json_object_new_object();
+						json_object_object_add(
+							json, "Neighbors",
+							json_neighbors);
+					}
+				}
+
+				for (ALL_LIST_ELEMENTS_RO(oi->neighbor_list, k,
+							  nbr)) {
+
+					if (!OSPF6_GR_IS_ACTIVE_HELPER(nbr))
+						continue;
+
+					if (!uj)
+						vty_out(vty,
+							" Neighbour %d :\n",
+							cnt++);
+
+					show_ospfv6_gr_helper_per_nbr(
+						vty, json_neighbors, uj, nbr);
+
+				}
+			}
+	}
+
+	return CMD_SUCCESS;
+}
+
+/* Graceful Restart HELPER  config Commands */
+DEFPY(ospf6_gr_helper_enable,
+      ospf6_gr_helper_enable_cmd,
+      "graceful-restart helper-only [A.B.C.D]",
+      "ospf6 graceful restart\n"
```

Comment:
nit: maybe reuse `OSPF_GR_STR` (available on master)

Reply:
I think , OSPF_GR_STR looks OSPF specific macro , do we need to add a separate macro for OSPF6 ? or do you still suggest to use these are one ?